torch/_tensor_str.py

```diff
@@ -65,132 +65,130 @@ def set_printoptions(
+                copy_abs[invalid_value_mask] = example_value
 
-def _number_format(tensor, min_sz=-1):
-    floating_dtype = tensor.dtype.is_floating_point  # save this because we cast later
-    _min_log_scale = _get_min_log_scale()
-    min_sz = max(min_sz, 2)
-    tensor = torch.DoubleTensor(tensor.size()).copy_(tensor).abs_().view(tensor.nelement())
+                exp_min = copy_abs.min()
+                if exp_min != 0:
+                    exp_min = math.floor(math.log10(exp_min)) + 1
+                else:
+                    exp_min = 1
+                exp_max = copy_abs.max()
+                if exp_max != 0:
+                    exp_max = math.floor(math.log10(exp_max)) + 1
+                else:
+                    exp_max = 1
 
-    pos_inf_mask = tensor.eq(float('inf'))
-    neg_inf_mask = tensor.eq(float('-inf'))
-    nan_mask = tensor.ne(tensor)
-    invalid_value_mask = pos_inf_mask + neg_inf_mask + nan_mask
-    if invalid_value_mask.all():
-        example_value = 0
-    else:
-        example_value = tensor[invalid_value_mask.eq(0)][0]
-    tensor[invalid_value_mask] = example_value
-    if invalid_value_mask.any():
-        min_sz = max(min_sz, 3)
-
-    int_mode = True
-    # TODO: use fmod?
-    for value in tensor.tolist():
-        if value != math.ceil(value):
-            int_mode = False
-            break
-
-    exp_min = tensor.min()
-    if exp_min != 0:
-        exp_min = math.floor(math.log10(exp_min)) + 1
-    else:
-        exp_min = 1
-    exp_max = tensor.max()
-    if exp_max != 0:
-        exp_max = math.floor(math.log10(exp_max)) + 1
-    else:
-        exp_max = 1
-    include_decimal_int_mode = floating_dtype and int_mode
-
-    scale = 1
-    exp_max = int(exp_max)
-    prec = PRINT_OPTS.precision
-    if int_mode:
-        if exp_max > prec + 1:
-            format = '{{:11.{}e}}'.format(prec)
-            fmt_fn = format.format
-            sz = max(min_sz, 7 + prec)
-        else:
-            sz = max(min_sz, exp_max + 1 + include_decimal_int_mode)
-            format = '{:' + str(sz) + '.0f}'
-            fmt_fn = format.format
-            if include_decimal_int_mode:
-                format = '{:' + str(sz - 1) + '.0f}'
-                nonfinite_format = format + '.'
-                fmt_fn = _get_format_fn(format, nonfinite_format)
-    else:
-        if exp_max - exp_min > prec:
-            sz = 7 + prec
-            if abs(exp_max) > 99 or abs(exp_min) > 99:
-                sz = sz + 1
-            sz = max(min_sz, sz)
-            format = '{{:{}.{}e}}'.format(sz, prec)
-            fmt_fn = format.format
-        else:
-            if exp_max > prec + 1 or exp_max < 0:
-                sz = max(min_sz, 7)
-                scale = math.pow(10, max(exp_max - 1, _min_log_scale))
-            else:
-                if exp_max == 0:
-                    sz = 7
+                if exp_max - exp_min > PRINT_OPTS.precision or exp_max > 8 or exp_min < -4:
```

Comment:
any specific rationale for these numbers?

Reply:
that's what I thought -- a comment to that effect would be helpful in case we forget.